src/Components/Components/src/Routing/Router.cs

```diff
@@ -55,6 +59,8 @@ public class Router : IComponent, IHandleAfterRender, IDisposable
         /// <inheritdoc />
         public void Configure(RenderHandle renderHandle)
         {
+            _logger = (ILogger<Router>)ServiceProvider.GetService(typeof(ILogger<Router>))
```

Comment:
Is the intent to make this optional in WASM? Could we add a comment to that effect? It's very tempting to optimize it and make this use property injection.

Reply:
Not really WASM (because that does have a logger by default) but further-out edge cases like unit test harnesses. Do you think there's any disadvantage to being tolerant to that?

Reply:
Nope, just that it stands out amongst other injected types.

Reply:
OK, I'm switching over to inject an `ILoggerFactory` since it's more conventional. The idea of making the logger optional doesn't really achieve anything.

Reply:
You can also inject `ILogger<Router>` directly which saves some annoying.